### ISSUE TITLE
10$: Error on React use* outside function

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#714]
```js
import { useState } from "react";
const [x, setX] = useState(0); // should error
```

**Comment by strager** [comment]
We'd need to do the following:
1. detect the "react" module and tag all named imports
2. for calls to functions named "use*" which are tagged as react imports, report an error if the use is in the global scope (including inside an `if` or `while` or whatever)

I don't know how we'd implement 1. Either the parser needs to communicate the module name to the linter, the parser needs to communicate the module tag to the linter, or the parser needs to communicate the React-use-function nature to the linter.

2 sounds straightforward.
